jobsdb/readonly_jobsdb.go

```diff
@@ -255,7 +259,12 @@ func (jd *ReadonlyHandleT) getUnprocessedJobsDSCount(ctx context.Context, ds dat
 
 	row := txn.QueryRow(sqlStatement)
 
-	defer txn.Commit()
+	defer func() {
+		err := txn.Commit()
```

Comment:
there is no point in committing a transaction which doesn't write to the database

Reply:
locks are released (beside after client disconnections, server shutdowns etc...) on both an explicit commit or rollback so there is a point on committing actually, but I agree that a rollback would more semantically correct in this case so happy to make the change.